pkg/services/ngalert/store/provisioning_store.go

```diff
@@ -0,0 +1,81 @@
+package store
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+type provenanceRecord struct {
+	Id         int `xorm:"pk autoincr 'id'"`
+	OrgId      int
+	RecordKey  string
+	RecordType string
+	Provenance models.Provenance
+}
+
+func (pr provenanceRecord) TableName() string {
+	return "provenance_type"
+}
+
+type ProvisioningStore interface {
+	GetProvenance(o models.ProvisionedObject) (models.Provenance, error)
+	// TODO: API to query all provenances for a specific type?
+	SetProvenance(o models.ProvisionedObject, p models.Provenance) error
+}
+
+func (st DBstore) GetProvenance(o models.ProvisionedObject) (models.Provenance, error) {
+	recordType := o.GetResourceTypeIdentifier()
+	recordKey := o.GetResourceUniqueIdentifier()
+
+	provenance := models.ProvenanceNone
+	err := st.SQLStore.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		result := make([]*provenanceRecord, 0)
+		q := "SELECT * FROM provenance_type WHERE record_key = ? AND record_type = ? ORDER BY id ASC LIMIT 1"
```

Comment:
I would recommend using XORM language rather than plain text queries to let XORM figure out SQL query specific to the database type